docs(Kommander): rename cloud providers to infrastructure providers (#2991)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.1.0-beta/operations/infrastructure-providers/configure-azure-infrastructure-provider/index.md

---
layout: layout.pug
navigationTitle: Configure an Azure Provider
title: Azure Infrastructure Provider with Static Credentials
excerpt: Configuring an Azure Infrastructure Provider
---

### Configuring an Azure infrastructure provider

> Before you configure an Azure Infrastructure Provider it's necessary to [install the Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)

Reviewer: ```suggestion
Before you configure an Azure Infrastructure Provider you must [install the Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
```

#### Create a new set of credentials using CLI commands

> To begin, you need a valid Azure account with credentials configured. You must be authorized as a Contributor in your Azure account, with the ability to assign roles to a user. The following commands create an active directory service principal, which you will delegate to Kommander, for creating Konvoy clusters:

Reviewer: ```suggestion
Before you begin, you need the following:

- A valid Azure account with configured credentials. 
- You must be authorized as a Contributor on your Azure account, with the ability to assign roles to a user. 

The following commands create an active directory service principal, which you delegate to Kommander, for creating Konvoy clusters:
```

```
az login
```

Determine the `SUBSCRIPTION_ID` of your account, named `id` in the output of the following command:

```
az account show
```
```json
{ "environmentName": "AzureCloud", "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX", "isDefault": true, "name": "ACME Enterprises Subscription", "state": "Enabled", "tenantId": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX", "user": { "name": "user@azureacme.onmicrosoft.com", "type": "user" } }
```

Create the service principal for the provider. Ensure that you replace `SUBSCRIPTION_ID` with the `id` from the previous command output.

```
az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/SUBSCRIPTION_ID"
```

The command returns data needed to create your infrastructure provider:

```
{
  "appId": "APP_ID",
  "displayName": "name",
  "name": "http://name",
  "password": "PASSWORD",
  "tenant": "TENANT"
}
```

> Next, you will assign the service principal the role of `User Access Administrator`. Replace `APP_ID` with the `appId` value from the output of the previous command:

Reviewer: ```suggestion
Next, assign the service principal the role of `User Access Administrator`. Replace `APP_ID` with the `appId` value from the output of the previous command:
```

```
az role assignment create --assignee "APP_ID" --role "User Access Administrator"
```

#### Fill out the Add Infrastructure Provider form

In Kommander, select the Workspace associated with the credentials you are adding.

Navigate to **Administration > Infrastructure Providers** and click the **Add Infrastructure Provider** button.

![Adding an Infrastructure Provider](/ksphere/kommander/1.1.0-beta/img/empty-infrastructure-providers.png)
<br />_Adding an Infrastructure Provider_

![Add Azure Infrastructure Provider Form](/ksphere/kommander/1.1.0-beta/img/add-azure-infrastructure-provider.png)
<br />_Add Azure Infrastructure Provider Form_

> - Enter a name for your infrastructure provider. Consider choosing a name that matches the Azure user.

Reviewer: ```suggestion
- Enter a name for your infrastructure provider. Select a name that matches the Azure user.
```

Reviewer 2: Again, I think it's a recommendation, not a requirement.
